test/resilience/tests/envoyproxy.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build resilience
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"github.com/envoyproxy/gateway/test/resilience/suite"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"testing"
+	"time"
+)
+
+func init() {
+	ResilienceTests = append(ResilienceTests, EPResilience)
+}
+
+var EPResilience = suite.ResilienceTest{
+	ShortName:   "EPResilience",
+	Description: "Envoyproxy resilience test",
+	Test: func(t *testing.T, suite *suite.ResilienceTestSuite) {
+		var ()
+
+		ap := kubernetes.Applier{
+			ManifestFS:     suite.ManifestFS,
+			GatewayClass:   suite.GatewayClassName,
+			ControllerName: "gateway.envoyproxy.io/gatewayclass-controller",
+		}
+
+		ap.MustApplyWithCleanup(t, suite.Client, suite.TimeoutConfig, "testdata/base.yaml", true)
+
+		t.Run("Envoy proxies continue to work even when eg is offline", func(t *testing.T) {
+			ctx := context.Background()
+
+			t.Log("Scaling down the deployment to 2 replicas")
+			err := suite.Kube().ScaleDeploymentAndWait(ctx, envoygateway, namespace, 2, time.Minute, false)
+			require.NoError(t, err, "Failed to scale deployment replicas")
+
+			t.Log("ensure envoy proxy is running")
+			err = suite.Kube().CheckDeploymentReplicas(ctx, envoygateway, namespace, 2, time.Minute)
+			require.NoError(t, err, "Failed to check deployment replicas")
+
+			t.Log("Scaling down the deployment to 0 replicas")
+			err = suite.Kube().ScaleDeploymentAndWait(ctx, envoygateway, namespace, 0, time.Minute, false)
+			require.NoError(t, err, "Failed to scale deployment to replicas")
+
+			t.Cleanup(func() {
+				err := suite.Kube().ScaleDeploymentAndWait(ctx, envoygateway, namespace, 1, time.Minute, false)
+				require.NoError(t, err, "Failed to restore replica count.")
+			})
+
+			require.NoError(t, err, "failed to add cleanup")
+
+			ns := "gateway-resilience"
+			routeNN := types.NamespacedName{Name: "backend", Namespace: ns}
+			gwNN := types.NamespacedName{Name: "all-namespaces", Namespace: ns}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			resultCh := make(chan error, 1)
+			go func() {
+				err := suite.Kube().CheckConnectivityJob(fmt.Sprintf("http://%s/welcome", gwAddr), 10)
+				resultCh <- err // Send the error (or nil) to the channel
```

Comment:
Is there some sort of envoy XDS cache invalidation (e.g. https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#ttl) that we should consider here to determine the time when requests should succeed before failure begin? 
@arkodg